libs/sdk-core/src/ldk/node_api.rs

```diff
@@ -209,8 +224,15 @@ impl NodeAPI for Ldk {
     async fn stream_incoming_payments(
         &self,
     ) -> NodeResult<Pin<Box<dyn Stream<Item = IncomingPayment> + Send>>> {
-        // TODO: Implement.
-        Ok(Box::pin(futures::stream::pending()))
+        let incoming_payments_rx =
+            self.incoming_payments_rx
+                .lock()
+                .await
+                .take()
+                .ok_or(NodeError::generic(
+                    "Incoming payments stream has already started",
+                ))?;
```

Comment:
What do you think about simplifying this by using a broadcast channel? No need for the option and the mutex on incoming_payments_rx. You would be able to subscribe multiple times if you want.